### ISSUE TITLE
Faulty Translation Files only throw a warning, not an error

### ISSUE THREAD

**Issue by michvllni** [microsoft/navcontainerhelper#1941]
**Describe the issue**
When there is an error in a translation file (for example an unescaped & or a missing closing tag), Compile-AppInBcContainer only throws a soft warning AL1206 and the compile is successful. If the compiled app file is published and installed, the translation file does not get applied at all.

```
warning AL1026: A warning occurred during XML validation: 'There is an error in XML document (233000, 27). An error occurred while parsing EntityName. Line 233000, position 27.
```

**Expected Behavior**
Compile-AppInBcContainer throws an error when there is an error in the translation file.

I've written a workaround for this by utilizing the outputTo parameter:
```powershell
Compile-AppInBcContainer [...] -outputTo {
                param($line)
                #throw translationfile errors as error even though they are just a warning
                if($line -like '*warning AL1026: A warning occurred during XML validation:*'){
                    Write-Host "##[error]$line" -ForegroundColor Red
                    throw "TranslationFileError"
                }
                elseif($line -like '*warning*'){
                    Write-Host "##[warning]$line" -ForegroundColor Yellow
                }elseif($line -like '*error*' -or ($line -like '*exception*')){
                    #this does not need to get thrown because it makes the app creation fail anyways
                    Write-Host "##[error]$line" -ForegroundColor Red
                }else{
                    Write-Host $line
                }
            }
```

Would it make sense to implement this into the default behavior?

**Comment by freddydk** [comment]
Thanks for the suggestion - will have a chat with Alex from the modernDev team about why this is just a warning.
Maybe I will add this behind a feature switch or parameter (to not risk breaking people).

**Comment by freddydk** [comment]
On the specific AL1026, the compiler shows this warning when an XML file cannot be read and while the compiler doesn't necessarily see this as an error, it is obvious that deleting some XML files might lead to errors.

I am thinking about introducing a generic concept like 

```
-treatWarningsLikeErrors @( "AL1026", "AL1048" )
```

or 

```
-treatWarningsLikeErrors @( "*" )
```

and then basically convert the warnings to errors.

On the other lines - ##[warning] etc - when you are using this code - you do not specify -azureDevOps right?

The -azureDevOps switch will convert warnings to stuff like ##vso[task.logissue type=warning;
I want to understand where ##[warning] comes from?

Thanks

**Comment by freddydk** [comment]
The next containerhelper preview should treat AL1026 as an error.

**Comment by michvllni** [comment]
Hi Freddy,
I didn't know that switch, I will try it out.
Currently I'm using ##[warning] and ##[error] to mark this as a warning/error in TFS 2017 OnPrem.
I'm still fairly new to all of this, so this was the first thing I found that worked

Your idea of the switch sounds like a good solution because teams can decide on their own what warning might be a breaking one for them.

Thank you!

**Comment by freddydk** [comment]
If you download the preview of ContainerHelper 2.0.13 - it has the new parameter.

**Comment by freddydk** [comment]
Shipped in 2.0.13
